Runtime/World/Component.h

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright(c) 2016-2019 Panos Karabelas
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+copies of the Software, and to permit persons to whom the Software is furnished
+to do so, subject to the following conditions :
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#pragma once
+
+//= INCLUDES =====================
+#include "../Core/EngineDefs.h"
+#include "Components/IComponent.h"
+#include <memory>
+#include <vector>
+#include <iostream>
+#include <functional>
+//================================
+
+namespace Spartan
+{
+    template <typename T>
+	struct SPARTAN_CLASS ComponentData
+	{
+		unsigned int mSize = 1;
+		std::array<std::shared_ptr<T>, 1024> mData;
```

Comment:
What about using a m_ prefix instead of an m prefix ?
This of course is a personal choice, but this projects happens to use the underscore version instead.
Consistency is key.